src/FSharp.Stats/Distributions/Distance.fs

```diff
@@ -0,0 +1,193 @@
+
+
+
+
+
+        /// Basic square function
+        let private square x = x * x
+
+        /// For value x, find index of the smallest value in sorted array which is larger than it
+        let private getCDFIndices (values : float []) x =
+            let rec searcher i =
+                if (i = values.Length) || x < values.[i]  then i 
+                else
+                    searcher (i+1)
+            searcher 0
+
+        /// Returns an array representing the cumulative sum beginning from the start of the array up to each element
+        let private cumulativeSum (values : float []) =
+            values
+            |> Array.mapFold (fun acc v -> 
+                                let s = acc + v
+                                s,s
+            ) 0.
+            |> fst
+
+        /// Returns the index mapping required to sort the values from small to large
+        let private getSortedIndices (values : float []) =
+            values
+            |> Array.indexed
+            |> Array.sortBy snd
+            |> Array.map fst
+
+        /// Sort given values by the order of the indices
+        let private sortByIndices indices (values : float []) =
+            indices 
+            |> Array.map (fun i -> values.[i])
+
+
+        /// Computes the normalized positions of allValues between the values of the distribution vs
+        let private getCDF (vs : float []) (allValues : float []) = 
+            let len = vs.Length |> float
+            let sorted = vs |> Array.sort 
+            let cdfIndices = 
+                allValues.[0 .. allValues.Length - 2]
+                |> Array.map (getCDFIndices sorted)
+
+            cdfIndices
+            |> Array.map (fun i -> (float i) / len)
+
+        /// Computes the weight-normalized positions of allValues between the values of the distribution vs
+        let private getWeightedCDF (vs : float []) weights (allValues : float []) = 
+            let len = vs.Length
+            let sortIndices = vs |> getSortedIndices
+            let sorted = sortByIndices sortIndices vs 
+            let cdfIndices = 
+                allValues.[0 .. allValues.Length - 2]
+                |> Array.map (getCDFIndices sorted)
+            let cumulativeWeights = 
+                weights
+                |> sortByIndices sortIndices 
+                |> cumulativeSum
+                |> Array.append [|0.|]
+            sortByIndices cdfIndices cumulativeWeights
+            |> Array.map (fun x -> x / cumulativeWeights.[len])
+
+
+        /// Computes the distance between the two cumulative distributions
+        let private computeDistanceOfCDFs p xCDFs yCDFs deltas = 
+            match p with
+            | 0 -> failwith "p can't be 0"
+            | 1 ->
+                (xCDFs,yCDFs,deltas)
+                |||> Array.map3 (fun xCDF yCDF delta ->
+                    xCDF - yCDF
+                    |> abs
+                    |> (*) delta   
```

Comment:
Please consider to preserve formula notation. For users looking for the math behind the implementation it is easier to search for formulas seen in e.g. publications.

The same applies to lines 99 and 111

Reply:
Tried to adress this in ccbc8bf. What do you think?